client/libra_wallet/src/mnemonic.rs

```diff
@@ -125,6 +238,51 @@ fn test_roundtrip_mnemonic() {
     assert_eq!(mnemonic.to_string(), other_mnemonic.to_string());
 }
 
+#[test]
+fn test_deterministic_mnemonic() {
+    let zeros_entropy: [u8; 32] = [0; 32];
+    let ones_entropy: [u8; 32] = [1; 32];
+
+    let zeros_mnemonic = Mnemonic::mnemonic(&zeros_entropy).unwrap();
+    let other_zeros_mnemonic = Mnemonic::mnemonic(&zeros_entropy).unwrap();
+    let ones_mnemonic = Mnemonic::mnemonic(&ones_entropy).unwrap();
+
+    let zeros_mnemonic_words = zeros_mnemonic.to_string();
+    let other_zeros_mnemonic_words = other_zeros_mnemonic.to_string();
+    let ones_mnemonic_words = ones_mnemonic.to_string();
+
+    assert_eq!(zeros_mnemonic_words, other_zeros_mnemonic_words);
+    assert_ne!(zeros_mnemonic_words, ones_mnemonic_words);
+}
+
+#[test]
+fn test_entropy_length() {
+    // entropy size in bytes.
+    for size in (16..32).step_by(4) {
+        let entropy = vec![0; size];
+        let mnemonic = Mnemonic::mnemonic(&entropy);
+        assert!(mnemonic.is_ok());
+    }
+
+    let some_invalid_entropy_sizes: [usize; 4] = [0, 8, 18, 36];
+    for size in some_invalid_entropy_sizes.iter() {
+        let entropy = vec![0; *size];
+        let mnemonic = Mnemonic::mnemonic(&entropy);
+        assert!(mnemonic.is_err());
+    }
+}
+
+#[test]
+fn test_entropy_to_word_number_compatibility() {
+    for size in (16..32).step_by(4) {
+        let entropy = vec![1; size];
+        let mnemonic = Mnemonic::mnemonic(&entropy).unwrap();
+        let mnemonic_string = mnemonic.to_string();
+        let mnemonic_from_string = Mnemonic::from(&mnemonic_string[..]);
+        assert!(mnemonic_from_string.is_ok());
+    }
+}
+
```

Comment:
Could we exercise the `Err` path returned by a failing checksum as well? I want tests to reflect there's a way this could go wrong outside entropy alignment.